### DISCUSSION TITLE
[VERY IMPORTANT!!!] Mechanism for converting solo plots into new protocol after May 17

### DISCUSSION THREAD

**Discussion by BiosNod** [Chia-Network/chia-blockchain#4955] [Ideas]
**VERY IMPORTANT!**

Hello developers, I watched the broadcast, which says that after May 17 a protocol for pools will be introduced, which will be possible to connect to pools, BUT one big BUT!

- will you have to replot for pool?
- for plots that you have made up to today - YES!
 
**Old plots created before the 17th can only be used in solo farming???**

In my opinion this is a factor undermining the credibility of the network and developers, **it is assumed that now there is NO mechanism for converting old plots into new ones after each implementation or adding changes to the protocol**, in the future every time the protocol changes it will be necessary to do replotting, which requires a LOT of electricity, **which makes farming NOT an eco-friendly coin**, only ensuring that the plots created at any time will not need to be redone **can make the chia a truly eco-friendly coin**. I consider it a mistake of the developers this missed moment, until we have some other mechanism.

I will not be able to mine in solo, but I have already killed nvme ssd to create solo plots and will I have to do them again if I want to use the pool protocol?

If you do not make a mechanism for converting old plots (already existing) into new with pool protocol, then farming will be useless for me and I will give my seed phrase to pools that will accept my old plots because now I have NO resources to sow again...

### I understand perfectly well that the old plots cannot support the new protocol, but can the existing plots be modified to support the new protocol? ###

Please support the topic and share this question so that the developers pay attention to this problem.

**Comment by TeddyKGBee** [comment]
I am in full agreement with this. I have asked several times the same question and have yet to receive a definitive answer. My question is: Is it technically possible to create some type of conversion tool for existing plot. Even if the tool required time to process each file and create any modifications that are necessary, that would be be a huge win for solo farmers. I understand that crating a tool like this may not fall high on priorities, but even if the developers could confirm or deny the ability to do this, it would provide some much needed hope for the future of farming and ultimate decentralization of the network, which is strongly needed with the rise of unofficial pools like Hpool. Additionally, it would generate a tremendous amount of good will with this of us that have supported this project from the begining.

**Comment by ilyakooo0** [comment]
> in the future every time the protocol changes it will be necessary to do replotting

This implies that after May 17 you will need to replot your existing plots (created before May 17). This is not accurate at all. All of the work you did as a solo farmer/plotter is still valid after May 17. Noone is invalidating anything you previously did. 

The only change is that there will be an additional option to create pool plots. This is backward-compatible with everything you did previously. It is a purely additive change.

**Reply by BiosNod** [reply]
I know that I can continue to use these solo blocks, but with the growth of the network, work and solo plots just turn into useless pumpkins in the beds, which will bloom in 10-100 years instead of 1 year when the difficulty was low)

In other words, the increase in the complexity of the network cancels all the work of a small solo miner can be considered completely useless, one can only hope for incredible luck or delete all of them and plot new pool-plots...

I'm not going to waste disk space with useless solo plots with incredibly low probability of winning, I'll have to replace them with new pool-plots and put up with wasted work(

**Reply by RedBrett** [reply]
Just my point-of-view here, but I think this explanation is theoretically accurate, but not practically accurate.  Anyone not farming on the scale of petabytes will a.) not have a dedicated large-scale plotting solution, b.) quickly find their win probability is laughably small and shrinking by the day, and ultimately c.) realize pooling will be the only way to realistically win anything.  Those individuals will most definitely need to eventually scrap their existing plots if they cannot be reused as pool plots, and re-create all new pool plots (especially if their storage for plots is limited).

**Reply by ilyakooo0** [reply]
Even if **everyone** nukes all of their plots and recreates them for pooling, the total energy will be minuscule compared to what Bitcoin burn every day.

I understand why you are upset. I don't buy the "greenness" argument.

**Comment by OversizedWizard** [comment]
Solo farming useless now anyway unless you permanently keep adding more storage.

Eco friendly mining my ass!!
Kills drives, uses copious amount of hard and CPU power.

It uses more electricity than a GPU miner.

And cost of buying drives and replacing faulty ones, for what, same odds as a lottery pmsl

**Comment by hotab** [comment]
I agree with this comment. It took me 2 months to fill up all my drives with plots, and probably 2 now semi-dead SSD drives. Now what you say is that I will have to go buy new ones to join pools.

IMHO - this is not credible solution, guys, and not backwards-compatible as claimed in the message.

**Reply by alexwwstx** [reply]
I believe the crazy space growth in the first several weeks after mainnet was released was totally out of the developer expectation, resulting in not so much time left to release pool ASAP

**Reply by hotab** [reply]
So? I am left with pretty much 2 dead SSDs, 2 months of time needed to replot, and pretty much 0 chia by this point due to crazy growth, and pretty much no way to solve this issue w/o throwing even more money to have even a small realistic chance to earn it all back.

I don't think I am the only one as well. I have seen network drop in size by 1 or 2 exabytes. I kinda think this would be the reason.

Eco friendliness argument also crumbles under this, btw. No time to do this protocol in this case is just poor excuse.

**Reply by alexwwstx** [reply]
I am in same position as you, plotted out 2000+ plots and now need to replot them again for joining a pool, otherwise, doesn't make any sense to hopelessly keep waiting for luck
an alternative solution is: join hpool if you trust it.

**Reply by hotab** [reply]
So - if I don't find how to solve this quickly and more or less cheaply - I will probably drop chia thanks to just this one thing. Frankly, I'd recommend every single person to do the same as next change might do the same thing to NFT plots.

Per hpool - I have very high ping to it, so it will probably be unstable.

Add to this that I also loose connection to chia network once every 1-2 days and have to manually bootstrap it.

**Reply by hotab** [reply]
Also, even if I do find a solution and keep farming it - I will not use it as a reliable coin (as it is obviously not considering just this discussion thread existing), and immediately sell.

**Comment by emlowe** [comment]
Conversion of plots from Original (OG) to Portable (PlotNFT) is not possible. Full Stop.

**Reply by reb0rn21** [reply]
You all have dozen of pools as Hpool, core, flex that mine with OG plots, NO ONE CARE of they are not official !!! why do even someone should say if you can or can not do something in decentralized project
So mine solo or use any old or new pool that support OG plost and do not listen to greedy new pool OPs that LIE you must replot (NO YOU DO NOT NEED TO REPLOT!!! even for pools!!!! just not the ones that use NFT)

**Comment by keliew** [comment]
FWIW, there are OG plot compatible pools out there...less than 5, and one of which don't require your mnemonics.

**Reply by hotab** [reply]
Unofficial pools have compatibility with OG plots, official ones don't. Anyways, as I have said in the above thread - this destroys trust in what is claimed in the this community completely, as I do remember reading that we will not need to replot for an arbitrary reason for the next few years at least. If 1 thing is not - then any other may not be as well.

**Comment by SlugFiller** [comment]
Sigh, let me unpack this. And forgive me for not at-ing every specific person making a statement here. The arguments boil down to the same things anyway, and each should hear all of them.

1. If you bought an SSD and it died from plotting, you should just RTM it. SSDs really shouldn't die like that.

2. I explained it in a different answer, but plots MUST have the target reward address irreversibly hard-coded into them as part of the plotting process. If there was some way to efficiently "convert" all the proofs within a plot, the same way could be used to "convert" proof you actually submitted, to proof belonging to someone else, thereby stealing the Chia you just farmed.

3. Originally, replotting would have had to have happened each time you changed plots. The use of "pool NFT" prevents that. Instead of complaining that you need to replot once due to the fact that plot NFTs were only just developed, you should be thankful that this is the ONLY time you have to replot, and you don't have to replot for every single time you switch between pools, as was the original plan.

4. From the get-go, Chia's mission plan did not aim for few people with excessively  large farms. On the contrary, copying statements from Burst, it pointed how being owned by a few large miners with dedicated machines is one of Bitcoin's major flaws, while PoC coins allow every Joe Blow with a bit of unused disk space on their documents drive to make a small plot and become a small influence on the network. It's the "democratization" of the coin, making it so that even a person without several grand to drop on a miner machine can be a small participant in the network. If you spent months plotting, you were probably going against the network's stated mission, the one that was public long before testnet was even in alpha.

5. Replotting was always going to be a thing. Even if pool NFTs allow you to switch pools without replotting (which is very important to ensure that a bad pool doesn't lock you in due to fear of replotting), there can be any number of events requiring replotting. Examples would be the minimum K-size going up, due to hardware improvements (from what I can tell, more plots with the minimum K-size are better than fewer plots with larger K, so it's better to stay with low K and replot later, then waste space in advance). A flaw being found in the algorithm requiring a change in proof format. The cryptography used by the public key being broken by quantum computers, requiring a new farmer and/or pool NFT public key. Replotting is something that should simply be expected every so often.

6. Despite this, this doens't make Chia "eco-unfriendly", because the amount of proof tests a plot goes through between potential replots is significantly larger the the amount of effort that goes into plotting it. The energy cost isn't 0, but it eventually pays for itself, compared to PoW.

7. I don't see the point about "credibility" at all. The very FIRST communication about Chia, when it was still just an idea, made it clear that few larger farms are against the purpose. Much of that communication was copied from Burst. Due to Bursts, then and now, lack of publicity, the importance of Hellman attack wasn't even mentioned in these earlier communications. Likewise, the FAQ STILL claims that changing pools would require plotting, even though that's no longer true, and you only need to replot once for the NFT. In other words, this statement was in the FAQ before pools were even a thing. It's not that the developers failed to communicate anything, it's that you failed to listen.

Most importantly, the sheer lack of understanding of how the network works required to ask for a "plot conversion" (see 2 above), is something one should not have when early-adopting experimental technology. And, like it or not, no matter how much marketing power there is behind it, Chia IS experimental. Mainnet has been up for less than 6 months, and pool support was only just released. And it's still missing vital features like hardware wallet support. Having to deal with growing pains is an early adopter's tax, and applies to all tech, not just crypto (Have you see what early VR kits looked like?). I know it can be enticing to walk in the same shoes as those who sold a pizza for 10,000 bitcoin, and could later buy a small country with it. But this also means walking in the same shoes of those who mined the early blocks, only to lose the pin code to the encrypted drive containing their private keys. Yes. Bitcoin as a predecessor allows Chia to avoid some problems, but it's still breaking new ground in many other ways. Either take it as it is, or get off the ride until it's ready for the mainstream.

**Reply by OversizedWizard** [reply]
Haven't you been reading the news????

Warranty on SSD drives has been reduced by manufacturers because of chia plotting!!!

You can't just RTM the drives. 
If you knew technology, then you'd know a drive, no matter what tech, has a finite number of writes.

Chia is a rip off. Simply put.  It's not being as green as it was meant too.
Thousand of drives dying, and millions of drives being added daily and keeps going.

That's not very green considering the carbon footprint to manufacture the drives.

**Reply by SlugFiller** [reply]
I actually DO know the technology, and know that a lot of the myth about writes is FUD. There was actually a study done on this, measuring the longevity of drives under different loads, and the load's influence on when the drive croaked was far less than the influence of its age. If your drive died young, it's not the usage, it was a built-in error. The study is called "Flash Reliability in Production: The Expected and the Unexpected", and one of its conclusions is "While wear-out from usage is often the focus of attention, we note that independently of usage the age of a
drive, i.e. the time spent in the field, affects reliability."

And, as someone already mentioned above, if everyone in the network nuked all their plots and replotted right now, the amount of energy used wouldn't be a fraction of what Bitcoin uses in a single day.

And yes, many drives are being added daily. It's not millions, at the moment. But when Chia becomes mainstream, it WILL be millions, because there's scarcely a household without 300-400GB of free diskspace somewhere, and the network, plus pools, make it a virtual no-loss to add one or two plots. The point is to add so many drives to the network that WEREN'T specifically purchased for it, that it no longer makes sense to purchase a drive for plots only. Then the cost of drives becomes irrelevant, because people are always going to buy drives for non-Chia reasons, while people would be discouraged to buy drives specifically for Chia.

The carbon footprint to manufacture drives for Chia would never be a drop in the ocean of the carbon footprint to manufacture drives for porn and mp3.

**Reply by OversizedWizard** [reply]
A house hold with 300-400 gif spare. Pmsl.

How long would it take to gain a reward using that much 

I had 32tb and never got a rewards.

And even reporting still uses resources and lower.

And for drives dying, you might be lucky, but they are dying. Fact. 

Nice short answer not the long winded qinge your doing pmsl.

Chia is shit and so not worth it.  Get over it.

**Reply by SlugFiller** [reply]
How long it would take to get a reward with 300-400 gigs? Depends on your pool. That's exactly the reason why pools are so important. If mining was limited to huge datacenters, then Chia WOULD be not worth it, as it would fail its main purpose.

The whole point of the network is many small miners, not few large miners. If you were picturing getting rich as the latter, then sorry, this network isn't for you. Come back when it becomes a mainstream currency for everyday products.

A lot of effort was put into Chia to ensure it attracts small everyday people, instead of being the realm of big speculative investors, few large-scale miners, and even fewer code-savvy enthusiasts. It's not worth it for people trying to game the system. It's totally worth it for people using it as intended.

**Reply by OversizedWizard** [reply]
Mate of mine, has 24tb of space.
And his pool rewards are so crap he be lucky to earn 1xch total in a year pmsl.

Even in a pool your rewards are based on your storage contribution and number of users on pool. That'll grow too making it useless lol.

So give an example of rewards in a pool for 1tb and proof.lol. 

When I asked mate what his rewards were, he quickly avoided the question and still haven't answered.
Because it's simply crap and not worth it.

I'm not in it to get rich quick, but I know when dedicating spare hard drive space can be a waste of time and chia is it unfortunately.
Spen a year and get 0.00000001 xch pmsl

Prove me wrong!!!

**Reply by keliew** [reply]
If it's not agreeable, why bother? Just. Move. On. Even if someone else can convince you, the hard numbers that you are fixated to is still truth in your eyes. So........

**Reply by SlugFiller** [reply]
Prove you wrong? Sure thing.

First, the growth of a pool is irrelevant. Unless the pool is running by dumb rules (*cough* PPLNS *cough), the decrease in per-block rewards due to pool growth is directly proportional to the increase in number of blocks obtained due to pool growth. In other words, it cancels out. The only thing that matters is your storage relative to the network as a whole.

A quick look at the stats states that the current farm rate is 22 years for 2XCH block for 1TB. So, mild number crunching brings that to about 0.00025 per day. That's a LOT more than 0.00000001 per year. And this is for 1 lousy TB. Even the minimum 0.1TB plot will earn 0.000025 per day. For comparison, a $6k+ ASIC Bitcoin miner will be lucky to earn 0.001BTC per day. Now think how many harddrives you could purchase for $6k, and how little electricity they would consume during plotting and a year of farming, compared to an ASIC running for a full year.

Eventually, the combination of the network growing, and hitting halving blocks, will result in the rewards gradually decreasing. This is fine. This is what happens in Bitcoin too, and is how it reached its current condition. This results in a coin's value increasing, and is why Bitcoin is worth ~$30k currently. If Chia was worth that right now, earning $7.5 a day, for doing absolutely nothing, and dedicating just 1 lousy TB in a drive you're barely using anyway, would be pretty crazy. That's why it isn't, obviously. And yet, at the same time, it's not as lower as one would expect. Unlike Bitcoin, it's not starting at 10k for a pizza. Heck, it was trading at $1k near the start. That's insane, given the farm-rate.
